File: shared/integration_example.go
package shared

import (
	"fmt"
	"kiki-agent-syncshield/schemas"
	"kiki-agent-syncshield/utils"
	// TODO: Update to correct module path for utils or use relative import if appropriate.
)

func ExampleUsage() {
	user := schemas.User{ID: "u1", Email: "test@example.com", Name: "Test User", IsActive: true}
	errors := utils.ValidateStruct(user)
	if len(errors) > 0 {
		fmt.Println("Validation failed:", errors)
	} else {
		fmt.Println("User valid:", user)
	}
	event := AuditEvent{Event: "login", UserID: user.ID, Data: map[string]interface{}{"ip": "127.0.0.1"}}
	fmt.Printf("Audit event: %+v\n", event)
}

File: shared/integration_example.py
from shared.audit_event import AuditEvent
from schemas.user import User
from utils.validation import validate_model

def example_usage():
    user_data = {"id": "u1", "email": "test@example.com", "name": "Test User"}
    user, errors = validate_model(User, user_data)
    if errors:
        print("Validation failed:", errors)
    else:
        print("User valid:", user)
    event = AuditEvent(event="login", user_id=user.id, data={"ip": "127.0.0.1"})
    print("Audit event:", event.__dict__)
